compiler-rt/cmake/Modules/AddCompilerRT.cmake

```diff
@@ -386,9 +386,14 @@ function(add_compiler_rt_runtime name type)
       target_link_libraries(${libname} PRIVATE ${builtins_${libname}})
     endif()
     if(${type} STREQUAL "SHARED")
-      if(COMMAND llvm_setup_rpath)
-        llvm_setup_rpath(${libname})
-      endif()
+## BEGIN SWIFT
+# Comment this out as it's being triggered somehow only for the last two
+# runtimes built, memprof and ubsan_minimal, and this doesn't make sense to run
+# simply because the llvm_setup_rpath() command happens to be defined sometimes.
+#      if(COMMAND llvm_setup_rpath)
```

Comment:
I reproduced this locally: my guess is that this isn't defined when most of the compiler-rt libraries are configured, but some invocation before the last two libraries calls that file, the function is interpreted, then it is defined here and adds the runpaths only for the last two libraries.

This whole setup of using the function if it is defined seems silly, and it is adding unwanted runpaths, so just commented it out for now.

Reply:
I've added @JDevlieghere since this is mostly a LLDB change. If this one is a real issue then it seems like it should be split and upstreamed.

The `lldb/tools/driver/CMakeLists.txt` also seems like it could be upstreamed?

Reply:
> But if this hunk is a real issue then it seems like it should be split and upstreamed.

It appears to be, I just downloaded and checked [the 15.0.6 build for linux x86_64](https://github.com/llvm/llvm-project/releases/tag/llvmorg-15.0.6) and it has these incorrect runpaths for all these compiler-rt libraries:
```
> find clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/ -name "lib*\.so*"|xargs readelf -d |ag "File:|runpath"|ag "/home/|/tmp/"
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.scudo_minimal.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.scudo.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.scudo_standalone.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.asan.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.memprof.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.hwasan_aliases.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.dyndd.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.ubsan_minimal.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.tsan.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.ubsan_standalone.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
File: clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/lib/clang/15.0.6/lib/x86_64-unknown-linux-gnu/libclang_rt.hwasan.so
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib:/tmp/llvm_release/final/Phase3/Release/llvmCore-15.0.6-final.obj/./lib]
```
I will submit a pull upstream later, but I'd like to get this in now to make sure this gets into the Swift 5.9 branch first.

> The lldb/tools/driver/CMakeLists.txt also seems like it could be upstreamed?

That change is not generally necessary if not building lldb separately from the rest of LLVM, as this Swift build does.